### ISSUE TITLE
running under alternate emacs.d?

### ISSUE THREAD

**Issue by brihadeesh** [Open-App-Library/Sanemacs#1]
Hi, I'm new to Emacs and wanted to give this a try since it looked considerably simpler than doom and a decent platform to build a custom config over. How could I go about setting this up in an alternate directory (other than `~/.emacs.d` that is). Should modifying the command displayed on the website work?

```
mkdir -p ~/.sanemacs.d; curl https://sanemacs.com/sanemacs.el > ~/.sanemacs.d/sanemacs.el
```

**Comment by DougBeney** [comment]
Hi! Thanks for trying it out.

You're on the right track with that command.

Now you just need to load the sanemacs.el file from wherever your main config is. So Emacs looks for `~/.emacs` or ~/.emacs.d/init.el` by default.

In either of those files you'd put a line like:

```elisp
(load "~/.sanemacs.d/sanemacs.el" nil t)
```

That "nil t" bit at the end I put there is just to suppress some annoying output that typically happens when you run the `load` function.

If you do not want to edit either of the above files, you can also do something like this to run emacs:

```bash
emacs -q --eval="(load \"/path/to/my-emacs-config.el\" nil t)"
```

The `-q` flag tells Emacs to not load the default user Emacs files.

One last thing you'll have to do. In Sanemacs, I set the custom file location to be right in ~/.emacs.d. You might want to change it to something else. The customize is where all of the Emacs customizer data is stored - ex. If you change your theme with `M-x customize-theme` and save.

(setq custom-file "~/.emacs.d/custom.el")

**Comment by brihadeesh** [comment]
Right this makes sense. I'd tried something similar some time back but had missed out on some of the elisp stuff you mentioned.

**Comment by brihadeesh** [comment]
That got things working! I'm using `chemacs` as well so that made things a hell of a lot easier and just added a profile for sanemacs.

So what I ended up doing eventually was to create a separate init file for sanemacs i.e. `~/.sanemacs.d/init.el` with this in it:

```
(load "~/.sanemacs.d/sanemacs.el" nil t)
```

and then added a profile for `chemacs` which pointed to the `~/.sanemacs.d` directory. So in effect `chemacs` handles launching sanemacs now. Although (like in doom-emacs) I'm not too keen on using the `customize` function for changing stuff, I've modified the line in sanemacs that sets the path to `custom.el`:

```
(setq custom-file "~/.sanemacs.d/custom.el")
```

Thanks a ton! Closing this now.
